index.js

```diff
@@ -11,11 +13,36 @@ server.get('/payments', (req, res) => {
     res.status(200).send(payments);
 });
 
-
+function validatePaymentDataMiddleWare(req, res, next) {
+    const payment = {...req.body};
+    const currentDate = new Date().toLocaleDateString('fr-CA');
+    
+    if(!(payment.date >= currentDate &&
```

Comment:
this is really an ugly way of implementing validation this way with all those `&&` conditions.
if validation fails, our api user would like to know what's gone wrong. so we should send that in our response, some sort of: 
```js
const validationErrors = validatePayment(payment);
// validatePayment should retun `null` or array of errors
if(validationErrors) {
  res.status(400).send( {errors: validationErrors} ); 
}
```
so, we can do it in much more clever way, like here: [validator-js](https://blog.logrocket.com/handle-data-validation-node-js-validatorjs/) or we can create our own validator following the [Decorator Pattern](https://refactoring.guru/design-patterns/decorator) as explained here [Validating Decorator](https://robdodson.me/posts/javascript-design-patterns-decorator)